routes/merged/project/location/firewall.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get api? do
+      firewall_list_api_response(firewall_list_dataset)
+    end
+
+    r.on NAME_OR_UBID do |firewall_name, firewall_id|
+      if firewall_name
+        r.post api? do
+          firewall_post(firewall_name)
+        end
+
+        filter = {Sequel[:firewall][:name] => firewall_name}
+      else
+        filter = {Sequel[:firewall][:id] => UBID.to_uuid(firewall_id)}
+      end
+
+      filter[:location] = @location
+      firewall = @project.firewalls_dataset.first(filter)
+
+      unless firewall
+        response.status = r.delete? ? 204 : 404
+        next
+      end
+
+      r.delete true do
+        Authorization.authorize(current_account.id, "Firewall:delete", firewall.id)
+        firewall.private_subnets.map { Authorization.authorize(current_account.id, "PrivateSubnet:edit", _1.id) }
+        firewall.destroy
+
+        if api?
+          response.status = 204
+          nil
+        else
+          {message: "Deleting #{firewall.name}"}
+        end
+      end
+
+      r.get true do
+        Authorization.authorize(current_account.id, "Firewall:view", firewall.id)
+        @firewall = Serializers::Firewall.serialize(firewall, {detailed: true})
+
+        if api?
+          @firewall
+        else
+          project_subnets = @project.private_subnets_dataset.where(location: @location).authorized(current_account.id, "PrivateSubnet:view").all
+          attached_subnets = firewall.private_subnets_dataset.all
+          @attachable_subnets = Serializers::PrivateSubnet.serialize(project_subnets.reject { |ps| attached_subnets.find { |as| as.id == ps.id } })
+
+          view "networking/firewall/show"
+        end
+      end
+
+      r.post %w[attach-subnet detach-subnet] do |action|
+        Authorization.authorize(current_account.id, "Firewall:view", firewall.id)
+
+        key = api? ? "private_subnet_id" : "private-subnet-id"
```

Comment:
I think we can rename the field in our web pages to `private_subnet_id` and eliminate this branch. This is what I did for previous merges into `Common`

Reply:
That makes sense to me.  I added a pull request for that.